ui/apps/platform/src/Containers/Policies/Wizard/Step1/MitreTechniqueSelect.tsx

```diff
@@ -24,52 +34,128 @@ function MitreTechniqueSelect({
         <Select
             aria-label={label}
             className={className}
             isOpen={isOpen}
+            selected={techniqueId}
             onSelect={onSelect}
-            onToggle={(_event, val) => setIsOpen(val)}
-            placeholderText={label}
-            selections={techniqueId}
-            maxHeight="300px"
+            onOpenChange={(nextOpen: boolean) => setIsOpen(nextOpen)}
+            toggle={toggle}
+            shouldFocusToggleOnSelect
         >
-            {mitreTechniques.map(({ id, name }) => {
-                const optionClassName = id.includes('.')
-                    ? 'mitre-subtechnique-option'
-                    : 'mitre-technique-option';
-                const indexOfColonSpace = name.indexOf(': ');
-
-                // See MitreAttackVectorsFormSection.css for name and id style rules.
-                return (
-                    <SelectOption
-                        key={id}
-                        value={id}
-                        isDisabled={getIsDisabledOption(id)}
-                        className={optionClassName}
-                    >
-                        <Flex
-                            flexWrap={{ default: 'nowrap' }}
-                            justifyContent={{ default: 'justifyContentSpaceBetween' }}
-                        >
-                            <span className="name">
-                                {indexOfColonSpace === -1
-                                    ? name
-                                    : name.slice(indexOfColonSpace + 2)}
-                            </span>
-                            <span className="id">{id}</span>
-                        </Flex>
-                    </SelectOption>
-                );
-            })}
+            <SelectList style={{ maxHeight: '300px', overflowY: 'auto' }}>
+                {Object.entries(groupedTechniques).map(([baseId, techniques], index, array) => {
+                    // Sort techniques within group: base technique first, then sub-techniques
+                    const sortedTechniques = techniques.sort((a, b) => {
+                        if (a.id === baseId) {
+                            return -1;
+                        }
+                        if (b.id === baseId) {
+                            return 1;
+                        }
+                        return a.id.localeCompare(b.id);
+                    });
+
+                    // Find the base technique for the group label
+                    const baseTechnique = sortedTechniques.find(
+                        (technique) => technique.id === baseId
+                    );
+                    const baseTechniqueName = baseTechnique ? baseTechnique.name : baseId;
+
+                    // Extract the part before the colon for the group label
+                    const indexOfColonSpace = baseTechniqueName.indexOf(': ');
+                    const groupLabel =
+                        indexOfColonSpace === -1
```

Comment:
Similar to the comment above. There are at least three instances of `indexOfColonSpace === -1` in the code when iterating over the techniques in order to gain information via string parsing. Could we handle grouping and sorting at the same time in a pure helper function, as well as create a data structure that allows direct access of things like the `groupLabel` and `displayName` so that the rendering code stays relatively simple?